forte/data/data_store.py

```diff
@@ -594,16 +655,65 @@ def _default_attributes_for_type(self, type_name: str) -> List:
         attr_dict: Dict = self._get_type_attribute_dict(type_name)
         attr_fields: Dict = self._get_entry_attributes_by_class(type_name)
         attr_list: List = [None] * len(attr_dict)
-        for attr_name, attr_id in attr_dict.items():
+        for attr_name, attr_info in attr_dict.items():
             # TODO: We should keep a record of the attribute class instead of
             # inspecting the class on the fly.
+            attr_id = attr_info[constants.ATTR_INDEX_KEY]
+
             attr_class = get_origin(attr_fields[attr_name].type)
```

Comment:
Right now we are still calling `self._get_type_attribute_dict(type_name)` and `self._get_entry_attributes_by_class(type_name)`. Can we get these info directly from `self._type_attributes` or `self._get_type_info()`?

Reply:
so `self._get_type_attribute_dict(type_name)` actually uses `self._get_type_info()`. With regards to `self._get_entry_attributes_by_class(type_name)`, we need this function because it gives us the `Field` object which us used to fetch the type information.